terraform/environments/data-platform/iam.tf

```diff
@@ -157,18 +157,35 @@ data "aws_iam_policy_document" "landing_to_raw_lambda_policy" {
       "${module.data_landing_s3_bucket.bucket.arn}",
     ]
   }
+
   statement {
-    sid    = "putRawData"
+    sid    = "copyToRaw"
     effect = "Allow"
     actions = [
-      "s3:PutObject*",
+      "s3:PutObject",
+      "s3:PutObjectTagging",
       "s3:ListBucket",
+
     ]
     resources = [
       "${module.data_s3_bucket.bucket.arn}/raw/*",
       "${module.data_s3_bucket.bucket.arn}/raw",
     ]
   }
+
+  statement {
+    sid    = "copyToFail"
+    effect = "Allow"
+    actions = [
+      "s3:PutObject",
+      "s3:PutObjectTagging",
+      "s3:ListBucket",
+    ]
+    resources = [
```

Comment:
LGTM, but I think you could combine the copyToRaw/copyToFail statements into one for simplicity since the actions are the same either way